src/Elastic.Apm.AspNetCore/Extensions/LoggerExtensions.cs

```diff
@@ -0,0 +1,26 @@
+using System;
+using Microsoft.Extensions.Logging;
+using LogLevel = Elastic.Apm.Logging.LogLevel;
+
+namespace Elastic.Apm.AspNetCore.Extensions
+{
+	internal static class LoggerExtensions
+	{
+		internal static LogLevel GetMinLogLevel(this ILogger logger)
+		{
+			if (logger == null) throw new ArgumentNullException(nameof(logger));
+
+			if (logger.IsEnabled(Microsoft.Extensions.Logging.LogLevel.Trace)) return LogLevel.Trace;
```

Comment:
I think we should replace `Level` property in `IApmLogger` by `IsEnabled` with the same semantics as `Microsoft.Extensions.Logging`'s `ILogger.IsEnabled`. This is the only way `IApmLogger.Level` is currently used anyway and it will avoid adding code like this. @gregkalapos WDYT?

Reply:
I don't really see the end result of it - we log the `LogLevel`, so we should know that (it's part of `void Log<TState>(LogLevel level, TState state, Exception e, Func<TState, Exception, string> formatter)`). But maybe it's just I don't see the whole thing here.

Reply:
I think there is some misunderstanding: log level that is included in the log record is derived from the level of the logging statement that emitted the record. For example, log records emitted by the following statement `logger.Debug()?.Log(...)` will have level `Debug`. OTOH `Level` property in `IApmLogger` controls which records are allowed to be emitted based on their level so it functions exactly like Microsoft.Extensions.Logging's `ILogger.IsEnabled`. Other logging libraries do it the same way as Microsoft.Extensions.Logging: for example [Serilog's `ILogger.IsEnabled`](https://github.com/serilog/serilog/blob/dev/src/Serilog/ILogger.cs#L190) and [NLog's `ILogger.IsTraceEnabled`](https://github.com/NLog/NLog/blob/master/src/NLog/ILogger.cs#L49)
Because of this discrepancy between the way Microsoft.Extensions.Logging does it and the way we currently specified `IApmLogger` interface (by forcing `Level` property), the bridge between `IApmLogger` and Microsoft.Extensions.Logging's `ILogger`, implemented in this PR, has to "guess" the minimal level that is not filtered out by repeatedly calling `ILogger.IsEnabled` as this code does. This is both wasteful from performance point of view and very hackish code because `IApmLogger.Level` is used only by [`LoggingExtensions.IfLevel`](https://github.com/elastic/apm-agent-dotnet/blob/master/src/Elastic.Apm/Logging/IApmLoggingExtensions.cs#L32) and it used exactly like `ILogger.IsEnabled`
```
logger.Level <= level ? new MaybeLogger(logger, level) : (MaybeLogger?)null;
```
It's like parsing a number from a string just to print it. But in this case it's even worse because it also prevents supporting dynamically changing the level. If we replace `IApmLogger`'s `Level` by `IsEnabled` then `AspNetCoreLogger.IsEnabled` from this PR can just call `_logger.IsEnabled` and automatically support more use cases (dynamically changing the level in particular) with less code - no need to guess level anymore and thus `LoggerExtensions.GetMinLogLevel` is not necessary.

Reply:
Ok, I see... so you want an `bool IsEnabled(LogLevel)` method (Or `bool IsDebugEnabled()`) methods - not an `bool IsEnabled` property.

So we'd have this:

```
public interface IApmLogger
{
	bool IsEnabled(LogLevel level);		
}
```

and this:
```
internal class AspNetCoreLogger : IApmLogger
{
   private readonly ILogger _logger;

   public bool IsEnabled(LogLevel level)
	=> _logger.IsEnabled(Convert(level));
//...more coee
}
```

Right?

Reply:
Yes, exactly. Although I'm not sure what you mean by `bool IsEnabled` property - do you mean NLog's approach with `bool IsDebugEnabled` etc. I'm not sure why they did that way (performance ?) - I would just go with Microsoft.Extensions.Logging and Serilog's `ILogger.IsEnabled` approach. It's hidden behind `LoggingExtensions.Trace()`, `LoggingExtensions.Debug()` etc. so even if in the future we change our minds and decide to go with NLog's `IsDebugEnabled` approach we will just need to change the way `LoggingExtensions.Trace()` is implemented but the rest of the agent won't be affected.

Reply:
> Although I'm not sure what you mean by bool IsEnabled property 

No.. that was my original interpretation of you original message, like I thought you want a property called `IsEnabled` that returns a bool - which was obviously not what you meant :) I think we understand each other now - C# code snippet helps ;)

Reply:
It would be nice to have consistent API with `Microsoft.Extensions.Logging`, however, I think it's not a goal of this PR. For me, it would be better to have another PR with replacing `Level` property to `IsEnabled`. @gregkalapos, @SergeyKleyman, what do you think about it?

Reply:
@vhatsura I would recommend to make this change in this PR but it's not a blocker for me so the final decision is yours.

The reason I recommend to make this change in this PR is because it's a classic case for refactoring - you want to implement a new feature but the current design is too rigid to accommodate it. Of course you can hack around the current design but I prefer the approach of refactoring the existing code and then implementing the new feature on top thus avoiding hacks. 

But in this case the refactoring required is very localized so postponing it doesn't matter that much that is why it's just a recommendation from me and the final decision is yours.

Reply:
I'd prefer to do it in another PR, mainly because it's a breaking change. I think having a breaking change as part of a feature PR isn't optimal - And since we can avoid it, it take that route.

Reply:
Got it. Will do changes in another PR